### PR TITLE
Fix DB2 integration tests DAT-13113

### PR DESCRIPTION
**Pull request by wwillard7800**
## Impact

- [x] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Description
The Db2IntegrationTest had issues

### CODE REVIEW

**Review by filipelautert — APPROVED** [review]

**Review by suryaaki2 — APPROVED** [review]

File: liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml
Comment (StevenMassaro) [inline liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml:1]: Are these changes necessary?

**Comment by StevenMassaro** [comment]
Nice work!! Question for you - is it possible to fix this in the testcontainers repo, rather than fix it here? I'm sure there are other people who would like to get these fixes.

**Comment by abrackx** [comment]
Wowowow I never thought I'd see the day! Nice work!

File: liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml
Comment (filipelautert) [reply liquibase-integration-tests/src/test/resources/changelogs/common/common.tests.changelog.xml:1]: They are required...

**Comment by filipelautert** [comment]
> Nice work!! Question for you - is it possible to fix this in the testcontainers repo, rather than fix it here? I'm sure there are other people who would like to get these fixes.

Good question! I think I can open an issue for them with my findings. It's quite simple - the code below makes it fail:

```
        Db2Container container = new Db2Container()
                .withUsername("lbuser")
                .withPassword("lbpass")
                .withDatabaseName("lbcat")
                .acceptLicense();

        List<PortBinding> portBindings = Collections.singletonList(new PortBinding(Ports.Binding.bindPort(50000), new ExposedPort(50000)));
        container.withCreateContainerCmdModifier(cmd -> cmd.withHostConfig(new HostConfig().withPortBindings(portBindings)));
        container.start();
```

If I exclude the` container.withCreateContainerCmdModifier` it works. We do it as part of our start method, but seems it is not required (at least for DB2) - and when it's done for DB2 it breaks the database startup. Also after finding it out I changed the PR again, now it's a lot more simple.

I'll create an issue at their github.

**Comment by filipelautert** [comment]
@StevenMassaro  https://github.com/testcontainers/testcontainers-java/issues/7679

File: liquibase-extension-testing/src/main/java/liquibase/extension/testing/testsystem/wrapper/DockerDatabaseWrapper.java [hunk 1]
```diff
@@ -58,7 +59,7 @@ public void start() throws Exception {
 
         container.withReuse(testSystem.getKeepRunning());
 
-        if (testSystem.getKeepRunning()) {
+        if (testSystem.getKeepRunning() && !(testSystem instanceof DB2TestSystem)) {
```
Comment (filipelautert) [inline liquibase-extension-testing/src/main/java/liquibase/extension/testing/testsystem/wrapper/DockerDatabaseWrapper.java:62]: @abrackx in fact the entire fix is here.

**Review by StevenMassaro — APPROVED** [review]
